source/module_elecstate/elecstate_pw.cpp

```diff
@@ -29,38 +29,38 @@ ElecStatePW<FPTYPE, Device>::~ElecStatePW()
     delmem_complex_op()(this->ctx, this->wfcr_another_spin);
 }
 
-template<typename FPTYPE, typename Device>
-void ElecStatePW<FPTYPE, Device>::init_rho_data() 
+template<typename T, typename Device>
+void ElecStatePW<T, Device>::init_rho_data() 
 {
     if (GlobalV::device_flag == "gpu" || GlobalV::precision_flag == "single") {
-        this->rho = new FPTYPE*[this->charge->nspin];
+        this->rho = new R*[this->charge->nspin];
```

Comment:
What is this 'R' here? Is it a bug or intended?

Reply:
@denghuilu I think this is good design for dealing with complex<real>/real , but I think the name of R and PossibleComplexToReal are a little bit confusing, do you mind to rename to "Real" and "GetTypeReal"

Reply:
Sure, I'll take care of it.